### ISSUE TITLE
Refactor and consolidate code for GPIO and HID keyword detectors

### ISSUE THREAD

**Issue by lucianomartin** [xmos/avs-device-sdk#73]
A considerable part of the code in the files:

 - https://github.com/xmos/avs-device-sdk/blob/master/KWD/GPIO/src/GPIOKeywordDetector.cpp
 - https://github.com/xmos/avs-device-sdk/blob/master/KWD/HIDsrc/HIDKeywordDetector.cpp

can be refactored and moved to a shared location. See a comment from @mbanth with more details:

```
I cannot agree with creating a separate keyword detector for HID and GPIO when writing in C++.
They share too much common functionality and have a common purpose.
The fact that they differ in the interface used to obtain the indices does not justify the duplication
given the capability that C++ provides to use a common interface to access differing implementations
of the same functionality.

I suggest that the I2C and USB specific functions should be taken out of the Keyword Detector class
and placed elsewhere.
Those functions that are identical for both interfaces, e.g., isAudioFormatCompatibleWithKW(),
use an abstract base class with a concrete function declaration and definition to provide the implementation.
Those functions that can share a common function signature but need different implementations,
e.g., deviceOpen(), use the same abstract base class with a pure virtual function declaration and
polymorphic derived class declarations and definitions to provide the necessarily different implementations.
```

**Comment by lucianomartin** [comment]
Changes included in https://github.com/xmos/avs-device-sdk/releases/tag/v1.25.0.3

**Comment by lucianomartin** [comment]
Fixed in #76
